contracts/0.4.24/StETH.sol

```diff
@@ -291,6 +300,29 @@ contract StETH is IERC20, Pausable {
         }
     }
 
+    /**
+     * @notice Moves `_sharesAmount` token shares from the caller's account to the `_recipient` account.
+     *
+     * @return a boolean value indicating whether the operation succeeded.
+     * Emits a `TransferShares` event.
+     * Emits a `Transfer` event.
+     *
+     * Requirements:
+     *
+     * - `_recipient` cannot be the zero address.
+     * - the caller must have at least `_sharesAmount` shares.
+     * - the contract must not be paused.
+     *
+     * @dev The `_sharesAmount` argument is the amount of shares, not tokens.
+     */
+    function transferShares(address _recipient, uint256 _sharesAmount) public returns (bool) {
```

Comment:
Imo, calling `transferShares(recipient, sharesAmt)` should lead to exactly the same outcome as calling `transfer(recipient, sharesAmt * sharePrice)`, meaning that both `transferShares` and `transfer` should emit `TransferShares` and `Transfer` events, the only difference being the measurement unit of the input amount.

Reply:
Also, I'd get rid of the return value: we use it in `transfer` since it's required by ERC20, but we (as well as most other token contracts out there) never actually return `false`, reverting the transaction instead. Keeping this return value here doesn't bring any advantages (since it's not a standard function) and at the same time either complicates the calling code or forces it to rely on the assumption that might possibly change.

Reply:
Instead, I'd return the number of stETH tokens transferred. We're already calculating this value to emit the `Transfer` event so it won't consume any additional gas.

Reply:
Makes sense; I'll update the code.
Thank you for coming here :+1:

Reply:
I think that we have a full resolution with the latest commits bunch :)